### DISCUSSION TITLE
`Listable` Type

### DISCUSSION THREAD

**Discussion by ilan-gold** [manzt/zarrita.js#172] [Q&A] [answered]
Hello! I'm trying to write a little `has` helper, something like:
```ts
export async function has(root: zarr.Group<FetchStore>, path: string) {
  try {
    await zarr.open(root.resolve(path));
  } catch (error) {
    if (error instanceof zarr.NodeNotFoundError) {
      return false;
    }
    return true;
  }
}
```
but the `root` could have as its type a `zarr.Group<Listable<FetchStore>>` since it can come from `tryWithConsolidated`.  But I can't annotate this function since [Listable is not exported](https://github.com/manzt/zarrita.js/blob/2779aba7b63cdc79c388f9ba8d6e7e54a5074d53/packages/core/src/consolidated.ts#L17-L20).  Any ideas?

**Comment by manzt** [comment, marked as answer]
I can work on exporting hat, but in the meantime you can use the `ReturnType` helper from TS to create a type alias:

```typescript
type MaybeListable = Awaited<ReturnType<typeof tryWithConsolidated>>;
```

**Reply by manzt** [reply]
See #176

**Reply by ilan-gold** [reply]
Thanks!
